File: bungee/src/main/java/live/turna/phenyl/bungee/listener/mirai/BungeeOnGroupMessageEvent.java
package live.turna.phenyl.bungee.listener.mirai;

import live.turna.phenyl.bungee.BungeePhenyl;
import live.turna.phenyl.bungee.event.BungeeGroupMessageEvent;
import live.turna.phenyl.common.eventhandler.mirai.OnGroupMessageEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;

/**
 * <b>BungeeOnGroupMessageEvent</b><br>
 * Listener on GroupMessageEvent.<br>
 * This produces all group messages including commands and chats.
 *
 * @since 2021/12/4 18:55
 */
public class BungeeOnGroupMessageEvent extends OnGroupMessageEvent<BungeePhenyl> implements Listener {
    public BungeeOnGroupMessageEvent(BungeePhenyl plugin) {
        super(plugin);
    }

    @EventHandler
    public void onGroupMessage(BungeeGroupMessageEvent e) {
        super.fill(e.getGroup(), e.getSenderId(), e.getMessage(), e.getSenderNameCardOrNick());
        super.handle();
    }
}

File: bungee/src/main/java/live/turna/phenyl/bungee/listener/mirai/BungeeOnGroupMessageEvent.kt
package live.turna.phenyl.bungee.listener.mirai

import live.turna.phenyl.bungee.BungeePhenyl
import live.turna.phenyl.bungee.event.BungeeGroupMessageEvent
import live.turna.phenyl.common.eventhandler.mirai.OnGroupMessageEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler

class BungeeOnGroupMessageEvent(plugin: BungeePhenyl) : OnGroupMessageEvent<BungeePhenyl>(plugin), Listener {
    @EventHandler
    fun onGroupMessage(e: BungeeGroupMessageEvent) {
        super.fill(e.group, e.senderId, e.message, e.senderNameCardOrNick)
        super.handle()
    }
}